ert_storage/endpoints/records.py

```diff
@@ -324,6 +324,67 @@ async def get_record_metadata(
     return bundle.metadata_dict
 
 
+@router.post("/ensembles/{ensemble_id}/records/{name}/observations")
+async def post_record_observations(
+    *,
+    db: Session = Depends(get_db),
+    ensemble_id: UUID,
+    name: str,
+    realization_index: Optional[int] = None,
+    observation_ids: List[UUID] = Body(...),
+) -> None:
+
+    if realization_index is None:
+        record_obj = _get_ensemble_record(db, ensemble_id, name)
+    else:
+        record_obj = _get_forward_model_record(db, ensemble_id, name, realization_index)
+
+    observations = (
+        db.query(ds.Observation).filter(ds.Observation.id.in_(observation_ids)).all()
+    )
+    if observations:
+        record_obj.observations = observations
+        flag_modified(record_obj, "observations")
+        db.commit()
+    else:
+        raise HTTPException(
```

Comment:
Maybe you could also include a test that tries to link observations that are not there?

Reply:
Should be ready now!